cmd/sink/main.go

```diff
@@ -132,9 +140,16 @@ func main() {
 	if err != nil {
 		logger.Fatalf("Could not connect to the database: %s\n", err)
 	}
-	sink := NewSink(db, logger)
+	filters, err := expr.NewFilters(os.Getenv(mountPathEnvVar))
+	if err != nil {
+		logger.Printf(
```

Comment:
If there are errors while loading our main feature, why we should load at all? If there are errors upon reading a change on the ConfigMap, then its another story.

Reply:
I disagree. Since the ConfigMap doesn't get created until users create a `KubeArchiveConfig`, it might be the case that this function returns an error because the ConfigMap has not been mounted when the sink starts. I'm working on logic right for `filters` to be able to update itself when files in the directory where the `ConfigMap` is mounted change. The filtering logic can handle the sink having no filters. It will log why it doesn't have any filters and it will log when it receives cloudevents but nothing will get archived or deleted

Reply:
The operator should create the `ConfigMap` if it doesn't exist as part of reconciliation.

But we should create the `ConfigMap` to make sure it's there, empty, at the start. I'll create an issue for this and add it to the operator.

The issue is https://github.com/kubearchive/kubearchive/issues/300.